src/main/java/com/hermesworld/ais/galapagos/topics/service/impl/TopicServiceImpl.java

```diff
@@ -178,13 +182,20 @@ public void init(KafkaCluster cluster) {
     @Override
     public CompletableFuture<Void> changeTopicOwner(String environmentId, String topicName,
             String newApplicationOwnerId) {
-        return doDeprecationAction(topicName, (kafkaCluster, metadata, eventSink) -> {
+        return doOnAllStages(topicName, (kafkaCluster, metadata, eventSink) -> {
+            if (metadata.getType() == TopicType.INTERNAL) {
+                return CompletableFuture
+                        .failedFuture(new IllegalStateException("Cannot change owner for internal topics"));
+            }
+            String previousOwnerApplicationId = metadata.getOwnerApplicationId();
+            List<String> producerList = new ArrayList<>(metadata.getProducers());
+            producerList.add(metadata.getOwnerApplicationId());
+            metadata.setOwnerApplicationId(newApplicationOwnerId);
+            producerList.remove(metadata.getOwnerApplicationId());
```

Comment:
This reads somewhat irritating (although it is correct). Better would be to remove `newApplicationOwnerId` here, would make code much more readable.